lib/MusicBrainz/Server/WebService/JSONSerializer.pm

```diff
@@ -14,53 +15,33 @@ sub fmt { 'json' }
 
 sub serialize
 {
-    my ($self, $type, @data) = @_;
+    my ($self, $type, $entity, $inc, $opts) = @_;
 
     $type =~ tr/-/_/;
 
-    my $override = $self->meta->find_method_by_name($type);
-    return $override->execute($self, @data) if $override;
+    if ($type =~ m/^(.*)_list$/ && exists($entity->{items})) {
+      if (exists($ENTITIES{$1}) && exists($ENTITIES{$1}->{plural_url})) {
+        $type = $1;
```

Comment:
1. Right, I still need to customize the emacs on my MB devel VM for that; it uses tabs by default.
2. Well until the exists checks succeed, $1 is just a potential entity type, so I kept it as $1 for the if and then assigned it to $type once it was verified to be a valid type. I have no issue with adding an $entity_type or $list_item_type or whatever to hold $1 right away. I'm not a huge fan of assigning the result of =~ (I think it reads better in the if), but if that's the standard in the MB codebase, then I'll use that.
Or I could drop the capture from the regex and use a substr inside the if to get the list item type; that avoids the $1 altogether (not sure if it would give better performance though).
3. No problem, I'll drop the m.